include/seqan3/range/container/concatenated_sequences.hpp

```diff
@@ -0,0 +1,1280 @@
+                 std::is_same_v<std::decay_t<rng_type>, std::decay_t<const_reference>> ||
+                 (input_range_concept<std::decay_t<rng_type>> &&
+                  std::is_same_v<std::decay_t<ranges::value_type_t<std::decay_t<rng_type>>>,
+                                 std::decay_t<ranges::value_type_t<std::decay_t<value_type>>>>)
+    {
+        return insert(pos, 1, std::forward<rng_type>(value));
+    }
+    // no specialisation for temporaries, since we have to copy anyway
+
+    /*!\brief Inserts count copies of value before position in the container.
+     * \tparam rng_type The type of range to be inserted; must be "like" `value_type`, i.e. satisfy
+     * seqan3::input_range_concept and have the same `value_type` as `value_type`.
+     * \param pos Iterator before which the content will be inserted. `pos` may be the end() iterator.
+     * \param count Number of copies.
+     * \param value Element value to insert.
+     * \returns Iterator pointing to the first element inserted, or pos if `count==0`.
+     *
+     * Causes reallocation if the new size() is greater than the old capacity(). If the new size() is greater
+     * than capacity(), all iterators and references are invalidated. Otherwise, only the iterators and
+     * references before the insertion point remain valid. The past-the-end iterator is also invalidated.
+     *
+     * \par Complexity
+     *
+     * Worst-case linear in concat_size(). This is a drawback over e.g. `std::vector<std::vector<alphabet>>`.
+     *
+     * \par Exceptions
+     *
+     * Basic exception guarantee, i.e. guaranteed not to leak, but container my contain bogus data after exceptions is
+     * thrown.
+     *
+     * \par Example
+     *
+     * ```cpp
+     * concatenated_sequences<dna4_vector> foobar;
+     * foobar.insert(foobar.end(), 2, "ACGT"_dna4);
+     * std::cout << foobar[0] << '\n'; // [A, C, G, T]
+     * std::cout << foobar[1] << '\n'; // [A, C, G, T]
+     * ```
+     */
+     template <typename rng_type>
+    iterator insert(const_iterator pos, size_type const count, rng_type const & value)
+        requires std::is_same_v<std::decay_t<rng_type>, std::decay_t<reference>> ||
+                 std::is_same_v<std::decay_t<rng_type>, std::decay_t<const_reference>> ||
+                 (input_range_concept<std::decay_t<rng_type>> &&
+                  std::is_same_v<std::decay_t<ranges::value_type_t<std::decay_t<rng_type>>>,
+                                 std::decay_t<ranges::value_type_t<std::decay_t<value_type>>>>)
+
+    {
+        auto const pos_as_num = std::distance(cbegin(), pos); // we want to insert BEFORE this position
+        // TODO SEQAN_UNLIKELY
+        if (count == 0)
+            return begin() + pos_as_num;
+
+
+        auto concatenated = ranges::view::repeat_n(value, count)
+                          | ranges::view::join
+                          | ranges::view::bounded;
```

Comment:
Mhmm would it be possible, to create this range and then call insert with pair of iterators for this range?
Could that reduce the implementation detail for two insert functions? Or is there a large runtime benefit to have both versions implemented separately?